### ISSUE TITLE
[Enhancement] Python submit failing to honor hfs=true

### ISSUE THREAD

**Issue by ddimatos** [ansible-collections/ibm_zos_core#376]
Having to use work around _submit to avoid this issue. This will need to be reversed when zoau has a fix. 
See file module utils [job.py](plugins/module_utils/job.py)
See lines [242](https://github.com/ansible-collections/ibm_zos_core/pull/348/files#diff-43f7641d31040832c1ee6d9297d962c8d025588af38b55987418737e8e4c42ddR242)
```
          # rstrip is a solution because we call zoau raw functions like _submit
            # which return values with a backslash
            job_id_stripped = entry.id.rstrip("\n")
```

See file [job_submit.py](plugins/modules/zos_job_submit.py)

Line [558](https://github.com/ansible-collections/ibm_zos_core/pull/348/files#diff-84a6bdbc151b19c5b1b7fe67ee5e6e9f1c1e3c0832ca1ce85173af04c57f1fb1R558)

```
# Imports datasets, ZOAUException, JobSubmitException are only needed for the
# ZOAU submit(hfs=True) bug
```

Line [582](https://github.com/ansible-collections/ibm_zos_core/pull/348/files#diff-84a6bdbc151b19c5b1b7fe67ee5e6e9f1c1e3c0832ca1ce85173af04c57f1fb1R582)
```
# Note that zoau submit will return None when wait to false, this is not desired
# so when wait = false, the raw version is called _submit.
```

Line [598](https://github.com/ansible-collections/ibm_zos_core/pull/348/files#diff-84a6bdbc151b19c5b1b7fe67ee5e6e9f1c1e3c0832ca1ce85173af04c57f1fb1R598)
```
        else:
            # If wait is not set, fallback to our 10 second default using _submit()
            job_listing = jobs._submit(src, None, **kwargs)
            jobId = job_listing.stdout_response.rstrip("\n")
            print("PRINT {0}".format(jobId))
```

Line [613](https://github.com/ansible-collections/ibm_zos_core/pull/348/files#diff-84a6bdbc151b19c5b1b7fe67ee5e6e9f1c1e3c0832ca1ce85173af04c57f1fb1R613)

```
# Note: this method has a work around resulting from ZOAU bug when hfs = true.
# For now, the JCL is put into a temporary data set and run from ZOAU which has
# no issues doing. One limitation of this work around is that users can not
# set the HLQ used in the temporary data set. The second issue is that is that
# in zoau jobs.submit is if wait is false, it immediately returns None which can
# in some cases make sense but in our code, if wait is not configured or puposely
# set to false, we wait up to 10 seconds so in this case the raw version _submit()
# had to be used and will need updating eventually when a fix is out. 
def submit_uss_jcl(src, module, timeout=0):
    """ Submit uss jcl. Use ZOAU jsub with option hfs=True """
```

line [628](https://github.com/ansible-collections/ibm_zos_core/pull/348/files#diff-84a6bdbc151b19c5b1b7fe67ee5e6e9f1c1e3c0832ca1ce85173af04c57f1fb1R628)
```
    tmp_data_set_for_submit = datasets.tmp_name(datasets.hlq())

    uss_copy_ds_rc = datasets.copy(src, tmp_data_set_for_submit)
    if uss_copy_ds_rc != 0:
        module.fail_json(
            msg="Error occurred while during job execution while copying jcl \
                  source {0} to {1}.".format(src, tmp_data_set_for_submit),
            rc=uss_copy_ds_rc,
            stdout=None,
            stderr="Non-zero return code received"
        )

    try:
        # Rearldess of wait boolean, if the timeout is set it takes precedence
        if timeout > 0:
            wait = True
            kwargs.update({"timeout": "{0}".format(timeout)})
            job_listing = jobs.submit(tmp_data_set_for_submit, wait, None, **kwargs)
            jobId = job_listing.id
        else:
            # Wait was false, rely on our 10 second default and use the _submit to do so
            job_listing = jobs._submit(tmp_data_set_for_submit, None, **kwargs)
            jobId = job_listing.stdout_response.rstrip("\n")
    except (ZOAUException, JobSubmitException) as err:
        module.fail_json(
            msg="Unable to submit job {0} as a result of {1}.".format(src, err),
            rc=None,
            stdout=None,
            stderr="Non-zero return code received."
        )
```

**Comment by ddimatos** [comment]
Corresponds to zoau 9225

**Comment by ddimatos** [comment]
This fix is in zoau build 1.2.1g, we need to validate by undoing the noted code here and work around.

**Comment by ddimatos** [comment]
Undoing the code revealed a new issue where randomly the job output failed to be retrieved with no valid explanation in the calls that go into the dependency, it took me a few days to  narrow down exactly the issue and how we can address and provide a work around. This is taking longer because its not easily reproducible, it is random and i believe it to be only in the development branch because of the migration from REXX to ZOAU. It can take 40 -50 job submissions to see this, and sometimes only 10, either way, i am in the process of cleaning up my tracing and changing the code a bit to be sent to pipeline for full regression.
